File: PaddleThroughLychnidos.Domain/Entities/ItineraryStop.cs
using PaddleThroughLychnidos.Domain.Shared;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PaddleThroughLychnidos.Domain.Entities
{
    public class ItineraryStop : IEntity
    {
        public int Id { get; set; }

        public int ItineraryId { get; set; }
        public Itinerary Itinerary { get; set; } = null!;

        public int ShopId { get; set; }
        public Shop Shop { get; set; } = null!;

        public int Order { get; set; }
        public string Notes { get; set; } = string.Empty;
        public TimeSpan SuggestedTime { get; set; }
    }
}

File: paddle-through-lychnidos-frontend/src/types/itineraryStop.ts
import type { TravelPlanShopSummary } from "./travelPlan";

// Mirrors PaddleThroughLychnidos.Application.Itinerary.Queries.ItineraryStopDto,
// returned inside GetByIdResponse.itinerary.stops by GET /api/itineraries/{id}.
// TimeSpan serializes as a "hh:mm:ss" string by default via System.Text.Json.
export interface ItineraryStop {
  order: number;
  notes: string;
  suggestedTime: string;
  shop: TravelPlanShopSummary;
}
